### ISSUE TITLE
Tests verifying that compilation with `--experimental-via-yul` produces same EVM assembly as two-stage compilation via Yul

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#11877]
The bytecode from `solc --bin --experimental-via-ir` should be identical to the one obtained by getting the output of `solc --ir` and then assembling it with `solc --strict-assembly`. Add some command line-tests to verify that.

Depends on  #11858 - this issue adds a command-line flag to disable printing Solidity snippets in EVM assembly.
Related to #11876. Until it's done, `--experimental-via-ir` flag must be added to `solc --ir` in order to get identical metadata.
Related to #11494.
#11875 contains the commands you could use (though the `--asm` output needs some processing to extract just the EVM assembly).

**Comment by cameel** [comment]
I added some tests in #12141. It does not test any multi-file contracts or files containing multiple contracts and @chriseth thinks it's important to include those as well so I'm going to do a followup PR to add them.
